cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java

```diff
@@ -190,14 +192,24 @@ private void addTimeRangeParametersToQuery(Map<String, String> timeRangeParams,
     }
   }
 
+  @Deprecated
   public RuntimeMetrics getFlowletMetrics(Id.Program flowId, String flowletId) {
-    return getMetrics(MetricsTags.flowlet(flowId, flowletId),
+    return getFlowletMetrics(flowId.toEntityId().flowlet(flowletId));
+  }
+
+  public RuntimeMetrics getFlowletMetrics(FlowletId flowletId) {
+    return getMetrics(MetricsTags.flowlet(flowletId),
                       Constants.Metrics.Name.Flow.FLOWLET_INPUT,
                       Constants.Metrics.Name.Flow.FLOWLET_PROCESSED,
                       Constants.Metrics.Name.Flow.FLOWLET_EXCEPTIONS);
   }
 
+  @Deprecated
   public RuntimeMetrics getServiceMetrics(Id.Program serviceId) {
+    return getServiceMetrics(serviceId.toEntityId());
+  }
+
+  public RuntimeMetrics getServiceMetrics(ProgramId serviceId) {
```

Comment:
Should we just call it get ``getProgramMetrics``, since it seems to apply to any program?